src/support.cpp

```diff
@@ -1479,30 +1479,30 @@ std::pair<Polygons, Polygons> AreaSupport::computeBasicAndFullOverhang(const Sli
     const Polygons outlines = mesh.layers[layer_idx].getOutlines();
     constexpr bool no_support = false;
     constexpr bool no_prime_tower = false;
-    const Polygons outlines_below = storage.getLayerOutlines(layer_idx - 1, no_support, no_prime_tower);
 
-    // To avoids generating support for textures on vertical surfaces, a moving average
-    // is taken over smooth_height. The smooth_height is currently an educated guess
-    // that we might want to expose to the frontend in the future.
     constexpr double smooth_height = 0.4; //mm
-    const auto layers_above = static_cast<LayerIndex>(std::round(smooth_height / mesh.settings.get<double>("layer_height")));
-    Polygons outlines_above;
-    for (int i = 0; layer_idx + i < mesh.layers.size() && i <= layers_above; ++i)
-    {
-        outlines_above.add(storage.getLayerOutlines(layer_idx + i, no_support, no_prime_tower));
-    }
-    outlines_above = outlines_above.unionPolygons();
+    const auto layers_below = static_cast<LayerIndex>(std::round(smooth_height / mesh.settings.get<double>("layer_height")));
 
     const coord_t layer_height = mesh.settings.get<coord_t>("layer_height");
     const AngleRadians support_angle = mesh.settings.get<AngleRadians>("support_angle");
     const double tan_angle = tan(support_angle) - 0.01; // The X/Y component of the support angle. 0.01 to make 90 degrees work too.
+    // overhang areas protruding less then `max_dist_from_lower_layer` don't need support
     const coord_t max_dist_from_lower_layer = tan_angle * layer_height; // Maximum horizontal distance that can be bridged.
 
+    // To avoids generating support for textures on vertical surfaces, a moving average
+    // is taken over smooth_height. The smooth_height is currently an educated guess
+    // that we might want to expose to the frontend in the future.
+    Polygons outlines_below = storage.getLayerOutlines(layer_idx - 1, no_support, no_prime_tower)
+                                                         .offset(max_dist_from_lower_layer);
+    for (int layer_idx_offset = 2; layer_idx - layer_idx_offset >= 0 && layer_idx_offset <= layers_below; layer_idx_offset ++)
+    {
+        auto outlines_below_ = storage.getLayerOutlines(layer_idx - layer_idx_offset, no_support, no_prime_tower)
```

Comment:
I'm sort-of fine with the name and structure as is, but if you know anything better for the temp-parameter name now would be a good time to change it.

Maybe just place it in there, since this is the only place it's needed, or, since `unionPolygons(...)` is symmetrical, maybe then swap the arguments as well so it's one flowing line. -- Although perhaps that would make the code _less_ readable instead of more.

Reply:
When programming I tried to resolve it by performing this statement in-line with in the `unionPolygons` function call below. Did not think that was an improvement over the current implementation.

I'm not a big fan of variable names like `outlines_below_temp` etc aswell, but couldn't really come up with a better name.